### ISSUE TITLE
Notifying change on URI results in a call to io.pivotal.arca.dispatcher.QueryListener, passing a cursor with the right count, but no data.

### ISSUE THREAD

**Issue by mathiastck** [cfmobile/arca-android#6]
We have an ArcaQuerySupportFragment

who's data is populated by calling:

final Query query = new Query(uri, QUERY_ID);
query.setForceUpdate(true);
query.setSortOrder(sort);
execute(query);

Another activity in our app updates the data and notifies the change:

ContentResolver contentResolver = context.getContentResolver();
contentResolver.update(uri, data, null, null);
contentResolver.notifyChange(uri, null);

our io.pivotal.arca.dispatcher.QueryListener is being called, and is passed a:
QueryResult result
whose:
result.getResult()
is a cursor with the right getCount(), and the right getColumnNames()

however after we call cursor.moveToPosition(), any calls to cursor.getString return null,
and any call to getInt returns 0.

So basically, we get a cursor back that looks right, except it has no data.

We have reproduced this issue against Arca 1.0-beta.3, as well as the latest dev branch code as of 7/28/2015.

**Comment by mathiastck** [comment]
Writing this out, I noticed our update call was not passing a where clause, so I think that was clearing our data.  I hit close on this issue.
